src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -533,26 +533,50 @@ private void RenderEndTagHelpersScope()
                                                   _tagHelperContext.ScopeManagerEndMethodName);
         }
 
-        private void RenderWriteTagHelperMethodCall(TagHelperChunk chunk)
+        private void RenderTagHelperOutput(TagHelperChunk chunk)
         {
+            var tagHelperOutputAccessor =
+                $"{ExecutionContextVariableName}.{_tagHelperContext.ExecutionContextOutputPropertyName}";
+
+            if (ContainsChildContent(chunk.Children))
```

Comment:
Discussed offline. There are some odd cases in Razor where useless / empty ParentChunk nodes are created and we need to avoid creating an empty `DefaultTagHelperContent` in those cases.

So, with a few more comments and the changes I suggested in `ContainsChildContent()`, I'm alright w/ this.